src/reward.ts

```diff
@@ -69,6 +69,12 @@ export default class Reward {
     return rewards.map(reward => new Reward(-reward.count, reward.type));
   }
 
+  /** For each reward, if it is of corresponding type, reduce the cost by dividing it by multiplier */
+  static discount(rewards: Reward[], type: Resource, multiplier: number): Reward[] {
```

Comment:
Sadly it seems this function is now unused

Reply:
Unless we use your suggestion above